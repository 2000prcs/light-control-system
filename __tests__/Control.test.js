import React from 'react';
import sinon from 'sinon';
import Controller from '../src/components/Control.jsx';

const controllerProps = {
  room: [
    {
      active: true,
      brightness: 50,
      id: 1,
      name: 'Balcony',
    },
  ],
  getCurrentRoom: () => {},
  getSwitchStatus: () => {},
};


describe('Controller component', () => {
  test('Should render Controller', () => {
    const ControllerWrapper = shallow(<Controller {...controllerProps} />);
    expect(ControllerWrapper).toMatchSnapshot();
  });

  test('Should toggle a switch when user clicks a light switch', () => {
    sinon.spy(Controller.prototype, 'toggleSwitch');
    const ControllerWrapper = mount(<Controller  {...controllerProps} />);
    const lightSwitch = ControllerWrapper.find('.switch');
    lightSwitch.first().simulate('click');
    expect(Controller.prototype.toggleSwitch.calledOnce).toBe(true);
    Controller.prototype.toggleSwitch.restore();
    ControllerWrapper.unmount();
  });
});
