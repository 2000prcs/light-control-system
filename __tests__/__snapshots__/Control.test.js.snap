// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controller component Should render Controller 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Control
    getCurrentRoom={[Function]}
    getSwitchStatus={[Function]}
    room={
      Array [
        Object {
          "active": true,
          "brightness": 50,
          "id": 1,
          "name": "Balcony",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Table
        columns={
          Array [
            Object {
              "field": "name",
              "label": "Room",
              "render": [Function],
              "sortable": true,
            },
            Object {
              "field": "active",
              "label": "State",
              "render": [Function],
              "sortable": false,
            },
            Object {
              "field": "brightness",
              "label": "Brightness",
              "render": [Function],
              "sortable": false,
            },
          ]
        }
        data={
          Array [
            Object {
              "active": true,
              "brightness": 50,
              "id": 1,
              "name": "Balcony",
            },
          ]
        }
        onRowClick={[Function]}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "columns": Array [
          Object {
            "field": "name",
            "label": "Room",
            "render": [Function],
            "sortable": true,
          },
          Object {
            "field": "active",
            "label": "State",
            "render": [Function],
            "sortable": false,
          },
          Object {
            "field": "brightness",
            "label": "Brightness",
            "render": [Function],
            "sortable": false,
          },
        ],
        "data": Array [
          Object {
            "active": true,
            "brightness": 50,
            "id": 1,
            "name": "Balcony",
          },
        ],
        "onRowClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Table
          columns={
            Array [
              Object {
                "field": "name",
                "label": "Room",
                "render": [Function],
                "sortable": true,
              },
              Object {
                "field": "active",
                "label": "State",
                "render": [Function],
                "sortable": false,
              },
              Object {
                "field": "brightness",
                "label": "Brightness",
                "render": [Function],
                "sortable": false,
              },
            ]
          }
          data={
            Array [
              Object {
                "active": true,
                "brightness": 50,
                "id": 1,
                "name": "Balcony",
              },
            ]
          }
          onRowClick={[Function]}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "columns": Array [
            Object {
              "field": "name",
              "label": "Room",
              "render": [Function],
              "sortable": true,
            },
            Object {
              "field": "active",
              "label": "State",
              "render": [Function],
              "sortable": false,
            },
            Object {
              "field": "brightness",
              "label": "Brightness",
              "render": [Function],
              "sortable": false,
            },
          ],
          "data": Array [
            Object {
              "active": true,
              "brightness": 50,
              "id": 1,
              "name": "Balcony",
            },
          ],
          "onRowClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
